ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/ContractKeysIT.scala

```diff
@@ -505,4 +505,65 @@ final class ContractKeysIT extends LedgerTestSuite {
       } yield ()
   })
 
+  test(
+    "CKGlocalKeyVisibility",
+    "Contract keys should be visible",
+    allocate(TwoParties),
+  )(implicit ec => { case Participants(Participant(ledger, alice, bob)) =>
+    for {
+
+      // create contracts to work with
+      cid <- ledger.create(alice, Test.WithKey(alice))
+
+      // double check its key can be found if visible
+      _ <- ledger.submit(
+        ledger.submitRequest(
+          alice,
+          Test.WithKey.key(alice).exerciseWithKey_NoOp(alice, alice).command,
+        )
+      )
+
+      // divulge the contract
+      helper <- ledger.create(bob, Test.WithKeyDivulgenceHelper(bob, alice))
+      _ <- ledger.exercise(alice, helper.exerciseWithKeyDivulgenceHelper_Fetch(_, cid))
+
+      // double check it is properly divulged
+      _ <- ledger.exercise(bob, cid.exerciseWithKey_NoOp(_, bob))
+
+      request = ledger.submitRequest(
+        bob,
+        // exercise by key the contract
+        Test.WithKey.key(alice).exerciseWithKey_NoOp(bob, bob).command,
+      )
+      failure1 <- ledger.submit(request).mustFail("exercise of a non visible key")
+
+      request = ledger.submitRequest(
```

Comment:
I feel like a single choice here that does both would probably be easier to understand than trying to do a multi-command submission but this looks fine afaict.